src/merge.ts

```diff
@@ -1,24 +1,114 @@
-import { IHttpHeaderParam, IHttpOperation, IMediaTypeContent, IServer } from '@stoplight/types';
+import {
+  IHttpOperation,
+  IHttpOperationRequestBody,
+  IHttpParam,
+  IMediaTypeContent,
+  INodeExample,
+  INodeExternalExample,
+  IServer,
+} from '@stoplight/types';
+import { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import { isEqual } from 'lodash';
 
-const mergeHeaders = mergeLists<IHttpHeaderParam[]>(
-  (h1, h2) => h1.name.toLowerCase() === h2.name.toLowerCase(),
-  h1 => h1, // ignore header #2 if mediaTypes is equal
-);
+type JSONSchema = JSONSchema4 | JSONSchema6 | JSONSchema7;
+
+function isSimpleAnyOfSchema(schema: JSONSchema) {
```

Comment:
we can also a type guard
```suggestion
function isSimpleAnyOfSchema(schema: JSONSchema): schema is { anyOf: JSONSchema[] } {
```

Reply:
👍